sdk/bpf/c/inc/solana_sdk.h

```diff
@@ -457,6 +457,37 @@ uint64_t sol_keccak256(
     uint8_t *result
 );
 
+
+/** Length of a secp256k1 recover input digest */
+#define SECP256K1_RECOVER_DIGEST_LENGTH 32
+/** Length of a secp256k1 input signature */
+#define SECP256K1_RECOVER_SIGNATURE_LENGTH 64
+/** Length of a secp256k1 recover result */
+#define SECP256K1_RECOVER_RESULT_LENGTH 64
+
+/** The digest provided to a sol_secp256k1_recover is invalid */
+#define SECP256K1_RECOVER_ERROR_INVALID_DIGEST 1
+/** The recovery_id provided to a sol_secp256k1_recover is invalid */
+#define SECP256K1_RECOVER_ERROR_INVALID_RECOVERY_ID 2
+/** The signature provided to a sol_secp256k1_recover is invalid */
+#define SECP256K1_RECOVER_ERROR_INVALID_SIGNATURE 3
+
+/**
+ * Recover public key from a signed message.
+ *
+ * @param digest Hashed message
```

Comment:
Might want to specify what mechanism must be sued to hash this message

Reply:
Ethereum uses Keccak256. But in general, any other mechanism will work too.